File: day12/py/puzzle1.py
#!/usr/bin/env python
from typing import TextIO, Tuple, List

from util import load_terrain, AStar
            
def solve():

    with open('../input.txt', 'r') as f:
        map, start, end = load_terrain(f)
        a_star = AStar(map)

    return a_star.solve(start, end)

if __name__ == "__main__":
    solution = solve()
    print(solution)

File: day12/cxx/puzzle1.cxx
#include <algorithm>
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

#include "util.hpp"

using namespace std;

const auto solve()
{
    ifstream f{"../input.txt"};

    auto terrain = load_terrain(f);

    AStar a_star {std::get<0>(terrain)};

    return a_star.solve(std::get<1>(terrain), std::get<2>(terrain));
}

int main()
{
    auto solution = solve();
    cout << solution << endl;
}